### PR TITLE
4.2 typings for db.migrations.operations

### PR DESCRIPTION
**Pull request by GabDug**
# I have made things!

Hello again!

Sorry to make such a big PR again, this one is focused on `django.db.migrations.operations`, but also improves some misc types, to make them on par with Django 4.2 

It removes 85 elements from the allowlist-todo.

### CODE REVIEW

File: django-stubs/db/migrations/operations/base.pyi [hunk 1]
```diff
@@ -13,6 +14,7 @@ class Operation:
     elidable: bool
     serialization_expand_args: list[str]
     _constructor_args: tuple[Sequence[Any], dict[str, Any]]
+    def __new__(cls, *args: Any, **kwargs: Any) -> Self: ...
```
Comment (adamchainz) [inline django-stubs/db/migrations/operations/base.pyi:17]: this seems kinda redundant? 🤔 I mean, I guess if stubtest says it's needed...

File: django-stubs/db/migrations/state.pyi [hunk 2]
```diff
@@ -7,8 +7,11 @@ from django.apps.registry import Apps
 from django.db.models.base import Model
 from django.db.models.fields import Field
 from django.db.models.manager import Manager
+from django.utils.functional import cached_property
 
-class AppConfigStub(AppConfig): ...
+class AppConfigStub(AppConfig):
+    def __init__(self, label: str) -> None: ...
+    def import_models(self) -> None: ...
```
Comment (adamchainz) [inline django-stubs/db/migrations/state.pyi]: This method exists in the base class with the same signature?
```suggestion
```

Comment (GabDug) [reply django-stubs/db/migrations/state.pyi]: Good catch thanks :)

This kind of testing would actually be very nice, to identify useless overrides!

**Review by adamchainz — APPROVED** [review]
Overall LGTM

File: django-stubs/db/migrations/operations/base.pyi (hunk 1 above)
Comment (GabDug) [reply django-stubs/db/migrations/operations/base.pyi:17]: Yep, stubtest complains, seems like the typeshed for object.__new__ is `def __new__(cls) -> Self: ...`

**Review by sobolevn — APPROVED** [review]
Great work! Thanks so much!

**Comment by adamchainz** [comment]
Awesome!

File: django-stubs/apps/config.pyi [hunk 3]
```diff
@@ -17,7 +18,8 @@ class AppConfig:
     models_module: str | None
     models: dict[str, type[Model]]
     def __init__(self, app_name: str, app_module: types.ModuleType | None) -> None: ...
-    default_auto_field: str
+    @cached_property
+    def default_auto_field(self) -> str: ...
```
Comment (intgr) [inline django-stubs/apps/config.pyi:22]: Heh this was just changed recently: https://github.com/typeddjango/django-stubs/pull/1541/files

Comment (intgr) [reply django-stubs/apps/config.pyi:22]: This looks like the cause of issue #1593

Comment (GabDug) [reply django-stubs/apps/config.pyi:22]: Should we ignore stubtest and set s `str`, or try to cover both with `default_auto_field: cached_property[str] | str `?

Comment (intgr) [reply django-stubs/apps/config.pyi:22]: Just `str` seems fine, overengineering seems more likely to cause issues.

**Pull request merged**